pkg/controller/reconciler/deployment_reconciler_test.go

```diff
@@ -446,6 +446,44 @@ func Test_DeploymentReconciler_Connections(t *testing.T) {
 	require.True(t, apierrors.IsNotFound(err))
 }
 
+// Creates a deployment with Radius disabled.
+//
+// Then checks the Events for Disabled-Disabled.
+func Test_DeploymentReconciler_RadiusDisabled_ThenRadiusDisabled(t *testing.T) {
+	ctx := testcontext.New(t)
+	_, client := SetupDeploymentTest(t)
+
+	name := types.NamespacedName{Namespace: "deployment-disabled-disabled", Name: "test-deployment-disabled-disabled"}
+	err := client.Create(ctx, &corev1.Namespace{ObjectMeta: ctrl.ObjectMeta{Name: name.Namespace}})
+	require.NoError(t, err)
+
+	deployment := makeDeployment(name)
+	err = client.Create(ctx, deployment)
+	require.NoError(t, err)
+
+	time.Sleep(time.Second * 5)
```

Comment:
Can you find a better way than using an explicit sleep? Sleeping for a fixed amount of time introduces bad tradeoffs in tests.

- The test can never go **faster** than the sleeps will allow, so you choose low numbers.
- Eventually you'll encounter a situation where the low number is not long enough, and then the test will be flaky.

PTAL at how the rest of these tests solve this with `require.Eventually`.